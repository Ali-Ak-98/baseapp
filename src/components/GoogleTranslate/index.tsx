import { History } from 'history';
import * as React from 'react';
// import CookieConsent from 'react-cookie-consent';
import { connect, MapDispatchToPropsFunction, MapStateToProps } from 'react-redux';
import {
    getCurrentLang,
    isSameLocale,
    triggerLanguageChange,
} from '../../helpers/googleTranslate';
import { languagesArray } from '../../helpers/googleTranslateLangs';
import {
    RootState,
} from '../../modules';

interface Locale {
    lang: string;
}

interface AppProps {
    history: History;
}

interface ReduxProps {
    locale: Locale;
}


type Props = AppProps & ReduxProps;


interface State {
    lang: string;
}


export class GoogleTranslateComponent extends React.Component<Props, State> {
    public unsubscribe?: () => void;
    public changed = false;
    public wrap = React.createRef<HTMLDivElement>();

    constructor(props) {
        super(props);
        this.state = {
            lang: '',
        };
    }

    public onChange = e => {
        if (this.wrap.current) {
            this.wrap.current.click();
        }
        triggerLanguageChange(e.target.value);
    };

    public componentDidMount() {
        window.addEventListener(
            'google-translate-lang-change',
            this.onGoogleLangChange,
        );
        // tslint:disable-next-line
        const lang = getCurrentLang();
        this.setState({
            lang: lang ? lang : '',
        });
    }

    public onGoogleLangChange = data => {
        const { detail } = data;
        const { lang } = this.state;
        if (detail !== lang) {
            this.setState({
                lang: detail,
            });
        }
    };

    public componentWillUnmount() {
        window.removeEventListener(
            'google-translate-lang-change',
            this.onGoogleLangChange,
        );
        this.unsubscribe && this.unsubscribe();
    }

    public onClick = e => {
        e.stopPropagation();
        e.nativeEvent.stopImmediatePropagation();
    };

    public render() {
        const { lang } = this.state;
        const { locale } = this.props;
        let languages = languagesArray;
        if (lang === '') {
            languages = [{ key: '', name: 'Translate' }, ...languagesArray];
        }

        if (isSameLocale(locale.lang)) {
            return null;
        }

        return (
            <div
                className="google-translate-widget"
                ref={this.wrap}
            >
                <select
                    onChange={this.onChange}
                    value={lang}
                    onClick={this.onClick}
                >
                    {languages.map(({ key, name }) => {
                        return <option value={key} key={key}>{name}</option>;
                    })}
                </select>
            </div>
        );
    }
}


const mapStateToProps: MapStateToProps<ReduxProps, {}, RootState> =
    (state: RootState): ReduxProps => ({
        locale: state.public.i18n,
    });

const mapDispatchToProps: MapDispatchToPropsFunction<{}, {}> =
    () => ({
    });

// tslint:disable-next-line:no-any
const GoogleTranslate = connect(mapStateToProps, mapDispatchToProps)(GoogleTranslateComponent) as any;

export {
    GoogleTranslate,
};

