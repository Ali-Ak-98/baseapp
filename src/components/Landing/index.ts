export * from './Header';
export * from './Launch';
