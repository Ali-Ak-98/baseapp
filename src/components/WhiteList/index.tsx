import classnames from 'classnames';
import * as React from 'react';
import { InjectedIntlProps, injectIntl } from 'react-intl';
import { connect, MapDispatchToProps } from 'react-redux';
import { TipIcon } from '../../assets/images/TipIcon';
import { TrashBin } from '../../assets/images/TrashBin';
import {
    beneficiariesDelete,
    Beneficiary,
    BeneficiaryBank,
    RootState,
    selectBeneficiaries,
    selectBeneficiariesCreate,
} from '../../modules';
import { WhiteListActivateModal } from './WhiteListActivateModal';
import { WhiteListAddModal } from './WhiteListAddModal';

interface ReduxProps {
    whitelist: Beneficiary[];
    whitelistAddData: Beneficiary;
}

interface DispatchProps {
    deleteAddress: typeof beneficiariesDelete;
}

interface OwnProps {
    currency: string;
    type: 'fiat' | 'coin';
    onChangeValue: (beneficiary: Beneficiary) => void;
}

interface State {
    currentWithdrawalBeneficiary: Beneficiary;
    isOpenAddressModal: boolean;
    isOpenConfirmationModal: boolean;
    isOpenDropdown: boolean;
    isOpenTip: boolean;
}

const defaultBeneficiary: Beneficiary = {
    id: 0,
    currency: '',
    name: '',
    state: '',
    data: {
        address: '',
    },
};

type Props = ReduxProps & DispatchProps & OwnProps & InjectedIntlProps;

// tslint:disable jsx-no-multiline-js
class WhiteListComponent extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            currentWithdrawalBeneficiary: defaultBeneficiary,
            isOpenAddressModal: false,
            isOpenConfirmationModal: false,
            isOpenDropdown: false,
            isOpenTip: false,
        };
    }

    public componentDidMount() {
        const { currency, whitelist } = this.props;
        if (currency && whitelist.length) {
            const filtredWhitelist = this.handleFilterByState(this.handleFilterByCurrency(whitelist, currency));
            if (filtredWhitelist.length) {
                this.handleSetCurrentAddress(filtredWhitelist[0]);
            }
        }
    }

    public componentWillReceiveProps(nextProps: Props) {
        const { currency, whitelist } = this.props;

        if ((nextProps.currency && nextProps.currency !== currency) ||
            (nextProps.whitelist.length && nextProps.whitelist !== whitelist)) {
            const filtredWhitelist = this.handleFilterByState(this.handleFilterByCurrency(nextProps.whitelist, nextProps.currency));

            if (filtredWhitelist.length) {
                this.handleSetCurrentAddress(filtredWhitelist[0]);
            }
        }
    }

    public render() {
        const {
            currency,
            type,
            whitelist,
            whitelistAddData,
        } = this.props;
        const {
            currentWithdrawalBeneficiary,
            isOpenAddressModal,
            isOpenConfirmationModal,
        } = this.state;
        const filtredWhitelist = this.handleFilterByState(this.handleFilterByCurrency(whitelist, currency));

        return (
            <div className="cr-white-list">
                <span className="cr-white-list__title">{type === 'coin' ? this.translate('page.body.wallets.whitelist.title') : this.translate('page.body.wallets.whitelist.fiat.title')}</span>
                {filtredWhitelist.length ? this.renderAddressDropdown(filtredWhitelist, currentWithdrawalBeneficiary, type) : this.renderAddAddress()}
                {isOpenAddressModal && (
                    <WhiteListAddModal
                        currency={currency}
                        type={type}
                        handleToggleAddAddressModal={this.handleToggleAddAddressModal}
                        handleToggleConfirmationModal={this.handleToggleConfirmationModal}
                    />
                )}
                {isOpenConfirmationModal && (
                    <WhiteListActivateModal
                        whitelistAddData={whitelistAddData}
                        handleToggleConfirmationModal={this.handleToggleConfirmationModal}
                    />
                )}
            </div>
        );
    }

    private renderAddAddress = () => {
        return (
            <div className="cr-white-list__add" onClick={this.handleClickToggleAddAddressModal()}>
                <span className="cr-white-list__add__label">{this.translate('page.body.wallets.whitelist.addAddress')}</span>
                <img src={require('../../assets/images/PlusIcon.svg')} className="cr-white-list__add__icon"/>
            </div>
        );
    }

    private renderDropdownItem = (item: Beneficiary, index: number, type: 'fiat' | 'coin') => {
        if (type === 'fiat') {
            return (
                <div key={index} className="cr-white-list__dropdown__body__item item">
                    <div className="item__left" onClick={this.handleClickSelectAddress(item)}>
                        <span className="item__left__title">{this.translate('page.body.wallets.whitelist.dropdown.fiat.name')}</span>
                        <span className="item__left__address">{item.name}</span>
                    </div>
                    <div className="item__left" onClick={this.handleClickSelectAddress(item)}>
                        <span className="item__left__title">{this.translate('page.body.wallets.whitelist.dropdown.fiat.fullName')}</span>
                        <span className="item__left__address">{item.data ? (item.data as BeneficiaryBank).full_name : ''}</span>
                    </div>
                    <div className="item__right">
                        <span className="item__right__delete" onClick={this.handleClickDeleteAddress(item)}><TrashBin/></span>
                    </div>
                </div>
            );
        }

        return (
            <div key={index} className="cr-white-list__dropdown__body__item item">
                <div className="item__left" onClick={this.handleClickSelectAddress(item)}>
                    <span className="item__left__title">{this.translate('page.body.wallets.whitelist.dropdown.address')}</span>
                    <span className="item__left__address">{item.data ? item.data.address : ''}</span>
                </div>
                <div className="item__right">
                    <span className="item__right__delete" onClick={this.handleClickDeleteAddress(item)}><TrashBin/></span>
                </div>
            </div>
        );
    }

    private renderDropdownBody = (whitelist: Beneficiary[], type: 'fiat' | 'coin') => {
        const dropdownBodyClassName = classnames('cr-white-list__dropdown__body', {
            'fiat-body': type === 'fiat',
        });

        return (
            <div className={dropdownBodyClassName}>
                {whitelist && whitelist.map((item, index) => this.renderDropdownItem(item, index, type))}
                <div className="cr-white-list__dropdown__body__add add" onClick={this.handleClickToggleAddAddressModal()}>
                    <span className="add__label">{this.translate('page.body.wallets.whitelist.addAddress')}</span>
                    <img className="add__icon" src={require('../../assets/images/PlusIcon.svg')}/>
                </div>
            </div>
        );
    }

    private renderDropdownTipCryptoNote = (note: string) => {
        return (
            <div className="tip__content__block">
                <span className="tip__content__block__label">{this.translate('page.body.wallets.whitelist.tipDescription')}</span>
                <span className="tip__content__block__value">{note}</span>
            </div>
        );
    }

    private renderDropdownTipCrypto = (currentWithdrawalBeneficiary: Beneficiary) => {
        if (currentWithdrawalBeneficiary) {
            return (
                <div className="cr-white-list__dropdown__tip tip">
                    <div className="tip__content">
                        <div className="tip__content__block">
                            <span className="tip__content__block__label">{this.translate('page.body.wallets.whitelist.tipAddress')}</span>
                            <span className="tip__content__block__value">{currentWithdrawalBeneficiary.data.address}</span>
                        </div>
                        <div className="tip__content__block">
                            <span className="tip__content__block__label">{this.translate('page.body.wallets.whitelist.tipName')}</span>
                            <span className="tip__content__block__value">{currentWithdrawalBeneficiary.name}</span>
                        </div>
                        {currentWithdrawalBeneficiary.description && this.renderDropdownTipCryptoNote(currentWithdrawalBeneficiary.description)}
                    </div>
                </div>
            );
        }

        return;
    }

    private renderDropdownTipFiatDescription = (description: string) => {
        return (
            <div className="tip__content__block">
                <span className="tip__content__block__label">{this.translate('page.body.wallets.whitelist.dropdown.fiat.description')}</span>
                <span className="tip__content__block__value">{description}</span>
            </div>
        );
    }

    private renderDropdownTipFiat = (currentWithdrawalBeneficiary: Beneficiary) => {
        if (currentWithdrawalBeneficiary) {
            return (
                <div className="cr-white-list__dropdown__tip tip fiat-tip">
                    <div className="tip__content">
                        <div className="tip__content__block">
                            <span className="tip__content__block__label">{this.translate('page.body.wallets.whitelist.dropdown.fiat.name')}</span>
                            <span className="tip__content__block__value">{currentWithdrawalBeneficiary.name}</span>
                        </div>
                        {currentWithdrawalBeneficiary.description && this.renderDropdownTipFiatDescription(currentWithdrawalBeneficiary.description)}
                        <div className="tip__content__block">
                            <span className="tip__content__block__label">{this.translate('page.body.wallets.whitelist.dropdown.fiat.account')}</span>
                            <span className="tip__content__block__value">{(currentWithdrawalBeneficiary.data as BeneficiaryBank).account_number}</span>
                        </div>
                        <div className="tip__content__block">
                            <span className="tip__content__block__label">{this.translate('page.body.wallets.whitelist.dropdown.fiat.bankOfBeneficiary')}</span>
                            <span className="tip__content__block__value">{(currentWithdrawalBeneficiary.data as BeneficiaryBank).bank_name}</span>
                        </div>
                    </div>
                </div>
            );
        }

        return;
    }

    private renderAddressDropdown = (whitelist: Beneficiary[], currentWithdrawalBeneficiary: Beneficiary, type: 'fiat' | 'coin') => {
        const { isOpenDropdown, isOpenTip } = this.state;

        const dropdownClassName = classnames('cr-white-list__dropdown', {
            'cr-white-list__dropdown--open': isOpenDropdown,
        });

        if (type === 'fiat') {
            return (
                <div className={dropdownClassName}>
                    <div className="cr-white-list__dropdown__select fiat-select select" onClick={this.handleToggleDropdown}>
                        <div className="select__left">
                            <span className="select__left__title">{this.translate('page.body.wallets.whitelist.dropdown.fiat.name')}</span>
                            <span className="select__left__address">{currentWithdrawalBeneficiary.name}</span>
                            <span className="select_left__title">{this.translate('page.body.wallets.whitelist.dropdown.fiat.fullName')}</span>
                            <span className="select__left__address">{currentWithdrawalBeneficiary.data ? (currentWithdrawalBeneficiary.data as BeneficiaryBank).full_name : ''}</span>
                        </div>
                        <div className="select__right">
                        <span className="select__right__tip" onMouseOver={this.handleToggleTip} onMouseOut={this.handleToggleTip}><TipIcon/></span>
                        <span className="select__right__select">{this.translate('page.body.wallets.whitelist.dropdown.select')}</span>
                        <span className="select__right__chevron"><img src={require('../../assets/images/ChevronIcon.svg')}/></span>
                        </div>
                    </div>
                    {isOpenDropdown && this.renderDropdownBody(whitelist, type)}
                    {isOpenTip && this.renderDropdownTipFiat(currentWithdrawalBeneficiary)}
                </div>
            );
        }

        return (
            <div className={dropdownClassName}>
                <div className="cr-white-list__dropdown__select select" onClick={this.handleToggleDropdown}>
                    <div className="select__left">
                        <span className="select__left__title">{this.translate('page.body.wallets.whitelist.dropdown.address')}</span>
                        <span className="select__left__address">{currentWithdrawalBeneficiary.data.address}</span>
                    </div>
                    <div className="select__right">
                    <span className="select__right__tip" onMouseOver={this.handleToggleTip} onMouseOut={this.handleToggleTip}><TipIcon/></span>
                    <span className="select__right__select">{this.translate('page.body.wallets.whitelist.dropdown.select')}</span>
                    <span className="select__right__chevron"><img src={require('../../assets/images/ChevronIcon.svg')}/></span>
                    </div>
                </div>
                {isOpenDropdown && this.renderDropdownBody(whitelist, type)}
                {isOpenTip && this.renderDropdownTipCrypto(currentWithdrawalBeneficiary)}
            </div>
        );
    }

    private handleClickDeleteAddress = (item: Beneficiary) => () => {
        this.handleDeleteAddress(item);
    }

    private handleClickSelectAddress = (item: Beneficiary) => () => {
        this.handleSetCurrentAddress(item);
    }

    private handleClickToggleAddAddressModal = () => () => {
        this.handleToggleAddAddressModal();
    }

    private handleDeleteAddress = (item: Beneficiary) => {
        const payload = {
            id: item.id,
        };

        this.props.deleteAddress(payload);
    }

    private handleFilterByCurrency = (whitelist: Beneficiary[], currency: string) => {
        if (whitelist.length && currency) {
            return whitelist.filter(item => item.currency.toLowerCase() === currency.toLowerCase());
        }

        return [];
    }

    private handleFilterByState = (whitelist: Beneficiary[]) => {
        if (whitelist.length) {
            return whitelist.filter(item => item.state.toLowerCase() === 'active');
        }
        return [];
    }

    private handleSetCurrentAddress = (item: Beneficiary) => {
        if (item.data) {
            this.setState({
                currentWithdrawalBeneficiary: item,
                isOpenDropdown: false,
            });
            this.props.onChangeValue(item);
        }
    }

    private handleToggleAddAddressModal = () => {
        this.setState(prevState => ({
            isOpenAddressModal: !prevState.isOpenAddressModal,
        }));
    }

    private handleToggleConfirmationModal = () => {
        this.setState(prevState => ({
            isOpenConfirmationModal: !prevState.isOpenConfirmationModal,
        }));
    }

    private handleToggleDropdown = () => {
        this.setState(prevState => ({
            isOpenDropdown: !prevState.isOpenDropdown,
        }));
    }

    private handleToggleTip = () => {
        this.setState(prevState => ({
            isOpenTip: !prevState.isOpenTip,
        }));
    }

    private translate = (id: string) => this.props.intl.formatMessage({ id });
}

const mapStateToProps = (state: RootState): ReduxProps => ({
    whitelist: selectBeneficiaries(state),
    whitelistAddData: selectBeneficiariesCreate(state),
});

const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = dispatch => ({
    deleteAddress: payload => dispatch(beneficiariesDelete(payload)),
});

// tslint:disable-next-line:no-any
export const WhiteList = injectIntl(connect(mapStateToProps, mapDispatchToProps)(WhiteListComponent) as any);
