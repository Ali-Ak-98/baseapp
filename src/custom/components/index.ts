export * from './DepositFiat';
export * from './BlockNationalityModal';
export * from './DepositFiat';
export * from './Dropdown';
export * from './History';
