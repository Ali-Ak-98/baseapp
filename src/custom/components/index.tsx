export * from './Deploy';
export * from './Order';
export * from './OrderForm';
export * from './RegisterInfo';
