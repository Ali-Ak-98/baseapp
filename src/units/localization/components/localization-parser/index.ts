import flatten, { unflatten } from 'flat';

import { LocalizationContainer } from '../../interfaces';
import { DEFAULT_LOCALIZATION_TEXT } from './default';

export const LocalizationParser = {
    parse: (jsonObject: any): LocalizationContainer => {
        const initObj = { ...DEFAULT_LOCALIZATION_TEXT };
        const initLines = flatten<{}, {}>(initObj, {});
        Object.keys(initLines).forEach((key) => {
            initLines[key] = `[${key}]`;
        });

        const localization = unflatten<any, any>({ ...initLines, ...jsonObject });
        return localization;
    },
};
