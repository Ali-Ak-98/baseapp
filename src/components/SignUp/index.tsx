import { Checkbox } from '@openware/components';
import cr from 'classnames';
import * as React from 'react';
import { Button } from 'react-bootstrap';
import { CustomInput } from '../';
import { EMAIL_REGEX, PASSWORD_REGEX } from '../../helpers';
import { PasswordValidationPopup } from './PasswordValidationPopup';

export interface PasswordValidationDetails {
    isLengthAcceptable: boolean;
    hasDigits: boolean;
    hasCapitalLetters: boolean;
    hasLowerCaseLetters: boolean;
}

export interface SignUpFormProps {
    isLoading?: boolean;
    title?: string;
    onSignUp: () => void;
    onSignIn?: () => void;
    className?: string;
    image?: string;
    labelSignIn?: string;
    labelSignUp?: string;
    emailLabel?: string;
    passwordLabel?: string;
    confirmPasswordLabel?: string;
    referalCodeLabel?: string;
    corporateTextLink?: string;
    termsMessage?: string;
    refId: string;
    password: string;
    email: string;
    confirmPassword: string;
    handleChangeEmail: (value: string) => void;
    handleChangePassword: (value: string) => void;
    handleChangeConfirmPassword: (value: string) => void;
    handleChangeRefId: (value: string) => void;
    hasConfirmed: boolean;
    clickCheckBox: () => void;
    validateForm: () => void;
    emailError: string;
    passwordError: string;
    passwordValidationDetails?: PasswordValidationDetails;
    confirmationError: string;
    handleFocusEmail: () => void;
    handleFocusPassword: () => void;
    handleFocusConfirmPassword: () => void;
    handleFocusRefId: () => void;
    confirmPasswordFocused: boolean;
    refIdFocused: boolean;
    emailFocused: boolean;
    passwordFocused: boolean;
    captchaType: 'recaptcha' | 'geetest' | 'none';
    renderCaptcha: JSX.Element | null;
    reCaptchaSuccess: boolean;
    geetestCaptchaSuccess: boolean;
    captcha_response: string;
}

export class SignUpForm extends React.Component<SignUpFormProps> {
    public render() {
        const {
            email,
            password,
            confirmPassword,
            refId,
            onSignIn,
            image,
            isLoading,
            labelSignIn,
            labelSignUp,
            emailLabel,
            passwordLabel,
            confirmPasswordLabel,
            referalCodeLabel,
            termsMessage,
            hasConfirmed,
            emailError,
            passwordError,
            passwordValidationDetails,
            confirmationError,
            emailFocused,
            passwordFocused,
            confirmPasswordFocused,
            refIdFocused,
            corporateTextLink,
        } = this.props;

        const emailGroupClass = cr('cr-sign-up-form__group', {
            'cr-sign-up-form__group--focused': emailFocused,
        });

        const passwordGroupClass = cr('cr-sign-up-form__group', {
            'cr-sign-up-form__group--focused': passwordFocused,
        });

        const confirmPasswordGroupClass = cr('cr-sign-up-form__group', {
            'cr-sign-up-form__group--focused': confirmPasswordFocused,
        });
        const refIdGroupClass = cr('cr-sign-up-form__group', {
            'cr-sign-up-form__group--focused': refIdFocused,
        });
        const termsLink = termsMessage ? (
            <div className="cr-sign-up-form__terms-link">
                {termsMessage.split(':')[0]} <a href="/terms">{termsMessage.split(':')[1]}</a>
            </div>
        ) : null;
        const logo = image ? (
            <h1 className="cr-sign-up-form__title">
                <img className="cr-sign-up-form__image" src={image} alt="logo" />
            </h1>
        ) : null;

        return (
            <form>
                <div className="cr-sign-up-form">
                    <div className="cr-sign-up-form__options-group">
                        <div className="cr-sign-up-form__option">
                            <div className="cr-sign-up-form__option-inner cr-sign-in-form__tab-signin" onClick={onSignIn}>
                                {labelSignIn ? labelSignIn : 'Sign In'}
                            </div>
                        </div>
                        <div className="cr-sign-up-form__option">
                            <div className="cr-sign-up-form__option-inner __selected">
                                {labelSignUp ? labelSignUp : 'Sign Up'}
                            </div>
                        </div>
                    </div>
                    <div className="cr-sign-up-form__form-content">
                        {logo}
                        <div className={emailGroupClass}>
                            <CustomInput
                                type="email"
                                label={emailLabel || 'Email'}
                                placeholder={emailLabel || 'Email'}
                                defaultLabel="Email"
                                handleChangeInput={this.props.handleChangeEmail}
                                inputValue={email}
                                handleFocusInput={this.props.handleFocusEmail}
                                classNameLabel="cr-sign-up-form__label"
                                classNameInput="cr-sign-up-form__input"
                                autoFocus={true}
                            />
                            {emailError && <div className="cr-sign-up-form__error">{emailError}</div>}
                        </div>
                        <div className={passwordGroupClass}>
                            <CustomInput
                                type="password"
                                label={passwordLabel || 'Password'}
                                placeholder={passwordLabel || 'Password'}
                                defaultLabel="Password"
                                handleChangeInput={this.props.handleChangePassword}
                                inputValue={password}
                                handleFocusInput={this.props.handleFocusPassword}
                                classNameLabel="cr-sign-up-form__label"
                                classNameInput="cr-sign-up-form__input"
                                autoFocus={false}
                            />
                            {!passwordValidationDetails && passwordError && <div className={'cr-sign-up-form__error'}>{passwordError}</div>}
                            {passwordValidationDetails && <PasswordValidationPopup visible={!!password} passwordError={passwordError} passwordValidationDetails={passwordValidationDetails} />}
                        </div>
                        <div className={confirmPasswordGroupClass}>
                            <CustomInput
                                type="password"
                                label={confirmPasswordLabel || 'Confirm Password'}
                                placeholder={confirmPasswordLabel || 'Confirm Password'}
                                defaultLabel="Confirm Password"
                                handleChangeInput={this.props.handleChangeConfirmPassword}
                                inputValue={confirmPassword}
                                handleFocusInput={this.props.handleFocusConfirmPassword}
                                classNameLabel="cr-sign-up-form__label"
                                classNameInput="cr-sign-up-form__input"
                                autoFocus={false}
                            />
                            {confirmationError && <div className={'cr-sign-up-form__error'}>{confirmationError}</div>}
                        </div>
                        <div className={refIdGroupClass}>
                            <CustomInput
                                type="text"
                                label={referalCodeLabel || 'Referral code'}
                                placeholder={referalCodeLabel || 'Referral code'}
                                defaultLabel="Referral code"
                                handleChangeInput={this.props.handleChangeRefId}
                                inputValue={refId}
                                handleFocusInput={this.props.handleFocusRefId}
                                classNameLabel="cr-sign-up-form__label"
                                classNameInput="cr-sign-up-form__input"
                                autoFocus={false}
                            />
                        </div>
                        <div style={{display: 'flex', alignItems: 'center'}}>
                        <Checkbox
                            checked={hasConfirmed}
                            className="cr-sign-up-form__checkbox"
                            onChange={this.props.clickCheckBox}
                            label={''}
                            // label={termsMessage ? termsMessage : 'I  agree all statements in '}
                        />
                            {termsLink}
                        </div>
                        {this.props.renderCaptcha}
                        <div className="cr-sign-up-form__button-wrapper">
                        <Button
                                block={true}
                                type="button"
                                disabled={this.disableButton()}
                                onClick={e => this.handleClick(e)}
                                size="lg"
                                variant="primary"
                            >
                                {isLoading ? 'Loading...' : (labelSignUp ? labelSignUp : 'Sign up')}
                            </Button>
                            <a className="cr-sign-up-form__corporate-link" key="ru" rel="alternate noopener noreferrer" href={'https://kb.emirex.com/corporate_account_verification'} hrefLang="ru" title={corporateTextLink} target="_blank">{corporateTextLink}</a>
                        </div>
                    </div>
                </div>
            </form>
        );
    }

    private disableButton = (): boolean => {
        const {
            email,
            password,
            confirmPassword,
            hasConfirmed,
            reCaptchaSuccess,
            isLoading,
            captchaType,
            geetestCaptchaSuccess,
        } = this.props;

        if (!hasConfirmed || isLoading || !email.match(EMAIL_REGEX) || !password || !confirmPassword) {
            return true;
        }
        if (captchaType === 'recaptcha' && !reCaptchaSuccess) {
            return true;
        }
        if (captchaType === 'geetest' && !geetestCaptchaSuccess) {
            return true;
        }
        return false;
    };

    private handleSubmitForm() {
        this.props.onSignUp();
    }

    private isValidForm() {
        const { email, password, confirmPassword } = this.props;
        const isEmailValid = email.match(EMAIL_REGEX);
        const isPasswordValid = password.match(PASSWORD_REGEX);
        const isConfirmPasswordValid = password === confirmPassword;

        return email && isEmailValid && (password && isPasswordValid) && (confirmPassword && isConfirmPasswordValid);
    }

    private handleClick = (e?: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
        if (e) {
            e.preventDefault();
        }
        if (!this.isValidForm()) {
            this.props.validateForm();
        } else {
            this.handleSubmitForm();
        }
    };
}
