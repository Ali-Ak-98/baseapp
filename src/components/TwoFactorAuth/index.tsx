import {Button} from '@openware/components';
import cr from 'classnames';
import {CustomInput} from '../';

import * as React from 'react';

export interface TwoFactorAuthProps {
    errorMessage?: string;
    isLoading?: boolean;
    show2faActions?: boolean;
    onSubmit: () => void;
    onDisable2FA: () => void;
    title: string;
    label: string;
    buttonLabel: string;
    message: string;
    otpCode: string;
    error: string;
    codeFocused: boolean;
    handleOtpCodeChange: (otp: string) => void;
    handleChangeFocusField: () => void;
    handleClose2fa: () => void;
    show2faActionsMsg: string;
    show2faActionsDisableLabel: string;
}

class TwoFactorAuthComponent extends React.Component<TwoFactorAuthProps> {
    public render() {
        const {
            errorMessage,
            isLoading,
            title,
            label,
            buttonLabel,
            message,
            error,
            otpCode,
            codeFocused,
            show2faActions,
            show2faActionsMsg,
            show2faActionsDisableLabel,
        } = this.props;

        const errors = errorMessage || error;
        const buttonWrapperClass = cr('cr-email-form__button-wrapper', {
            'cr-email-form__button-wrapper--empty': !errors,
        });
        const emailGroupClass = cr('cr-email-form__group', {
            'cr-email-form__group--focused': codeFocused,
        });
        return (
            <div className="pg-2fa___form">
                <form>
                    <div className={show2faActions ? 'cr-email-form withActions' : 'cr-email-form'}>
                        <div className="cr-email-form__options-group">
                            <div className="cr-email-form__option">
                                <div className="cr-email-form__option-inner">
                                    {title || '2FA verification'}
                                    <div className="cr-email-form__cros-icon" onClick={this.handleCancel}>
                                        <img alt="" src={require('../EmailForm/close.svg')}/>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="cr-email-form__form-content">
                            {show2faActions &&
                            <div className="actionsWrapper">
                                <div className="message">
                                    <img alt="warning" src={require('../../assets/images/warning.svg')}/>
                                    <p>{show2faActionsMsg}</p>
                                </div>
                                <hr/>
                                <div className="buttons">
                                    <button onClick={this.handleDisable2FA}
                                            type="button">{show2faActionsDisableLabel}</button>
                                </div>
                            </div>
                            }
                            <div className="cr-email-form__header">
                                {message}
                            </div>
                            <div className={emailGroupClass}>
                                <CustomInput
                                    type="number"
                                    label={label || '6-digit Google Authenticator Code'}
                                    placeholder={label || '6-digit Google Authenticator Code'}
                                    defaultLabel="6-digit Google Authenticator Code"
                                    handleChangeInput={this.props.handleOtpCodeChange}
                                    inputValue={otpCode}
                                    handleFocusInput={this.props.handleChangeFocusField}
                                    classNameLabel="cr-email-form__label"
                                    classNameInput="cr-email-form__input"
                                    onKeyPress={this.handleEnterPress}
                                    autoFocus={true}
                                />
                                {errorMessage && <div className="cr-email-form__error">{errorMessage}</div>}
                            </div>
                            <div className={buttonWrapperClass}>
                                <Button
                                    label={isLoading ? 'Loading...' : (buttonLabel ? buttonLabel : 'Sign in')}
                                    className={otpCode ? 'cr-email-form__button' : 'cr-email-form__button cr-email-form__button--disabled'}
                                    disabled={isLoading || !otpCode.match(/.{6}/g)}
                                    onClick={this.handleSubmit}
                                />
                            </div>
                        </div>
                    </div>
                </form>
            </div>
        );
    }

    private handleCancel = () => {
        this.props.handleClose2fa();
    };

    private handleSubmit = () => {
        this.props.onSubmit();
    };


    private handleDisable2FA = () => {
        this.props.onDisable2FA();
    };

    private handleEnterPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
        if (event.key === 'Enter') {
            event.preventDefault();
            this.handleSubmit();
        }
    };
}

export const TwoFactorAuth = TwoFactorAuthComponent;
