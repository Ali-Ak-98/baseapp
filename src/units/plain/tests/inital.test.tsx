import * as React from 'react';
import { shallow } from 'enzyme';
import { JestTestComponentProps, JestTestComponent } from '../components/test';

const defaultProps: JestTestComponentProps = {
    text: 'Hello world!',
    title: '',
};

const wrapper = (props: Partial<JestTestComponentProps> = {}) => {
    return shallow<JestTestComponentProps>(<JestTestComponent {...defaultProps} {...props} />);
};

describe('Initial', () => {
    it('case 1', () => {
        expect(wrapper({ title: 'tooltip' })).toMatchSnapshot();
    });
});
