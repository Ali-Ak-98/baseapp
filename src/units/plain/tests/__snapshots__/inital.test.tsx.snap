// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial case 1 1`] = `
<div
  title="tooltip"
>
  Hello world!
</div>
`;
